qutip/bloch.py

```diff
@@ -393,12 +393,12 @@ def add_states(self, state, kind='vector', colors=None, alpha=1.0):
 
             if kind == 'vector':
                 if colors is not None:
-                    self.add_vectors(vec, colors=colors[k], alpha=alpha)
+                    self.add_vectors(vec, colors=[colors[k]], alpha=alpha)
```

Comment:
Would this break code of those who already implemented the list of list method?
Could we support both, or add a custom error message to update them to the change?

Reply:
It will break the list of list method but [according to the doc,](https://qutip.readthedocs.io/en/qutip-5.0.x/apidoc/classes.html#qutip.bloch.Bloch.add_states), it should have never worked like this.

But I agree that it can be convenient to add a check in the method

Reply:
I have added a check and I have cleaned the function as asked in #1917